plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -173,9 +175,32 @@ internal class LocationPuckManager(
     vararg bearings: Double,
     options: (ValueAnimator.() -> Unit)? = null,
     forceUpdate: Boolean = false
+  ) {
+    if (settings2.puckBearingEnabled) {
+      animationManager.setEnabled(true)
```

Comment:
just mentioned one thing that concerns me here: `updateCurrentBearing` is technically getting called very often not depending if `puckBearingEnabled` is true or false. That actually means that e.g. when `puckBearingEnabled = false` - we will still be calling 0-duration animation each time and also `animationManager.setEnabled(false)`. Similarly  `animationManager.setEnabled(true)` is called each time.
Can we actually incapsulate that logic and perform it only when `puckBearingEnabled` did actually change?
The only thing we truly need to do on each bearing update is `animateToBearing(bearings, options, forceUpdate)` when `settings2.puckBearingEnabled = true`

Reply:
Resolved by filtering out calls to animateToBearing when puckBearingEnabled is false.